### PR TITLE
Add og:title and :description to lessons. Maybe.

### LINKED ISSUE #6: Quote Requested: Lesson Builder Meta Data
From Jeff via [basecamp](https://radcampaign.basecamphq.com/projects/13482568-pulitzer-monthly-maintenance/posts/99601358/comments):
> Can you provide us with a quote on fixing the issue with the Lesson Builder where Google doesn't display any actual Lesson Builder content in its search results and Facebook doesn't pull content into posts -- often the only summary we get in either is Lesson Builder | Pulitzer Center Lesson Builder.

### PR DESCRIPTION
**Pull request by xaqrox**
A few things going on here.

1. Tidy up some logic in `lessonCtrl.js` to make sure all nooks and crannies are checked for metatag data.
2. Add [`angular-update-meta`](https://github.com/jvandemo/angular-update-meta#usage) `og:*` tags to `lessonView.php`, populated with the same data as their corresponding regular metatags.
3. Add vague default `og:*` tags to `lesson-builder-html.tpl.php`, as per `angular-update-meta`'s README:

> If the meta element does not exist yet, it is NOT added, so make sure they exist in your original head element.

The question of whether the fact that this is an angular app is preventing crawlers from getting the meta information is an open one, because I am a n00b :baby:. Not sure how to test that empirically w/o deploying this to production. @mkly do you have any insight on this?

Closes #6, hopefully.

### CODE REVIEW

File: sites/all/modules/custom/lesson_builder/app/components/lesson/lessonCtrl.js [hunk 1]
```diff
@@ -52,29 +54,47 @@ app.controller('LessonController', function ($scope, $routeParams, Lesson, User,
 
     // See if a lesson has custom meta tags for description and title, if not, add
     // defaults, for use with angular-update-meta.
-    if($scope.lessonData.metatags) {
+    if ($scope.lessonData.metatags) {
       // See if custom meta description exists, add body if not
-      if($scope.lessonData.metatags['und'].hasOwnProperty("description")) {
-        $scope.lessonData.metatags.description = $scope.lessonData.metatags['und']['description']['value'];
-      } else {
-        if($scope.lessonData.body.length) {
```
Comment (xaqrox) [inline sites/all/modules/custom/lesson_builder/app/components/lesson/lessonCtrl.js]: It was this check that was preventing the description from getting into the markup. `@scope.lessonData.body` is an Object, so `.length` is always undefined, even if there is data in there, so we never pull in the body content. I'm not really sure how `$scope.lessonData.metatags` is put together, so I can't explain why the metatags are not being filled in with tokens like usual. However if explicit (non-token) metatags are entered, they do show up.

**Comment by jselt** [comment]
credentials in lastpass to get a sql dump?

**Comment by mkly** [comment]
```
    <meta property="og:image" content="http://placekitten.com/1200/630" />
    <meta property="og:site_name" content="Pulitzer Center" />
    <meta property="og:type" content="website" />
    <meta property="og:title" content="Pulitzer Center Lesson Builder" />
    <meta property="og:description" content="Pulitzer Center Lesson Builder description goes here" />
```
og

**Comment by xaqrox** [comment]
So, good news, then?

**Comment by mkly** [comment]
I was just writing. That is just the base content. I looked that the ticket and it reads
```
Can you provide us with a quote on fixing the issue with the Lesson Builder where Google doesn't display any actual Lesson Builder content in its search results and Facebook doesn't pull content into posts -- often the only summary we get in either is Lesson Builder | Pulitzer Center Lesson Builder.  
```
This means all content from all the urls. Which is most likely not doable because of how Facebook and Google crawl unless we build out some isomorphic for them.

**Pull request merged**